phoenix-core/src/main/java/org/apache/phoenix/compile/JoinCompiler.java

```diff
@@ -517,8 +597,24 @@ public JoinType getType() {
             return onConditions;
         }
 
-        public JoinTable getJoinTable() {
-            return joinTable;
+        public JoinTable getRhsJoinTable() {
```

Comment:
add many methods in JoinSpec for OnNodeVisitor to make the modification of internal states of  JoinSpec is sealed in JoinSpec class.